src/systems/battery_plugin/LinearBatteryPlugin.cc

```diff
@@ -395,13 +425,20 @@ void LinearBatteryPluginPrivate::OnDisableRecharge(
   this->startCharging = false;
 }
 
+//////////////////////////////////////////////////
+void LinearBatteryPluginPrivate::OnBatteryDrainingMsg(
+  const char *, const size_t, const ignition::transport::MessageInfo &)
+{
+  this->startDrainingFromTopics = true;
+}
+
 //////////////////////////////////////////////////
 void LinearBatteryPlugin::PreUpdate(
   const ignition::gazebo::UpdateInfo &/*_info*/,
   ignition::gazebo::EntityComponentManager &_ecm)
 {
   IGN_PROFILE("LinearBatteryPlugin::PreUpdate");
-  this->dataPtr->startDraining = false;
+  this->dataPtr->startDraining = this->dataPtr->startDrainingFromTopics;
```

Comment:
This means that once draining has started from a topic, it will never stop draining. However if draining starts as a result of joint commands, the vehicle coming to a stop also stops the drainage. This particularly relevant for recharging since it will likely not be possible to recharge if a vehicle starts draining batteries via a topic.

If we're okay with this, I recommend we say so in the documentation of `<power_draining_topic>`.

Reply:
Hmm, maybe for the case of tracked vehicles I could expose the desired track velocity as a component of the link. Something like `components::TrackVelocity`. Then the change required here would be much smaller.

However, I still think it might be useful to have the option to start draining via a topic. If we'd go all the way down to ign-msgs, we could even create a message for starting/stopping the battery draining.

Reply:
I'm okay leaving this as a todo item. I've added notes and documentation in f88afb0.

Reply:
Leaving it as a todo sounds good to me.